torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,271 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data"""
+import glob
+import os
+from typing import Any, Callable, Dict, List, Optional, cast
+
+from rasterio.crs import CRS
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.colors import ListedColormap
+
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import NonGeoDataset, RasterDataset
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class L7Irish(RasterDataset):
+    r"""L7 Irish dataset.
+
+    The `L7 Irish <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
+    Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes, displayed in the biomes listed below. Manually generated cloud masks 
+    are used to train and validate cloud cover assessment algorithms, which in turn are intended to compute the percentage of 
+    cloud cover in each scene.
+
+    Dataset features:
+
+    * 206 scenes from Landsat-7 Enhanced Thematic Mapper Plus (ETM+) tiles
+    * Imagery from global tiles between June 2000 - December 2001
```

Comment:
```suggestion
    * Imagery from global tiles between June 2000--December 2001
```
En dash instead of hyphen to denote a range